lib/src/testing/test_file.dart

```diff
@@ -0,0 +1,188 @@
+import 'dart:io';
+import 'dart:isolate';
+
+import 'package:path/path.dart' as p;
+
+import '../../dart_style.dart';
+
+final _indentPattern = RegExp(r'\(indent (\d+)\)');
+final _fixPattern = RegExp(r'\(fix ([a-x-]+)\)');
+final _unicodeUnescapePattern = RegExp(r'×([0-9a-fA-F]{2,4})');
+final _unicodeEscapePattern = RegExp('[\x0a\x0c\x0d]');
+
+/// Get the absolute local file path to the package's "test" directory.
+Future<String> findTestDirectory() async {
+  var libraryUri = await Isolate.resolvePackageUri(
+      Uri.parse('package:dart_style/src/testing/test_file.dart'));
+  return p.normalize(
+      (p.join(p.dirname(libraryUri!.toFilePath()), '../../../test')));
```

Comment:
[nit] looks like one extra set of parens

```suggestion
      p.join(p.dirname(libraryUri!.toFilePath()), '../../../test'));
```